api/src/main/java/org/apache/iceberg/actions/GenerateSymlinkManifest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+/**
+ * An action that generates a symlink manifest table for an Iceberg table.
+ *
+ * <p>This action will generate symlink manifests for a table. A symlink manifest is a file with
+ * links to the data files in the table. Systems which do not integrate with Iceberg can read the
+ * underlying data in the Iceberg tables as an external table. The symlink manifest table will
+ * expose all the hidden partitions as the union of historical partition specs.
+ */
+public interface GenerateSymlinkManifest
+    extends Action<GenerateSymlinkManifest, GenerateSymlinkManifest.Result> {
+
+  /**
+   * Set a root location for the symlink manifest table. If a root location is not provided the
+   * default location will be at table_root/_symlink_format_manifest/snapshot_id/
+   *
+   * @param rootLocation root location for symlink manifest table
+   * @return this for method chaining
+   */
+  GenerateSymlinkManifest rootLocation(String rootLocation);
```

Comment:
Nit / non-blocking:

1) From reading the JavaDoc, I'm not sure if the `rootLocation` function is the prefix and then we'll still append `/${snapshot_id}/`.

2) Arguably that's an implementation detail but But I’m also not sure how `rootLocation` should be passed in. For example, for a table at `s3://bucket/database/table`, if I wanted the symlink manifest to be at `s3://bucket/database/table/current_symlink/`, what would I need to pass in?

It’s possible that both of these are more of a question for the implementing class, but at least for the second question it would be nice when using Iceberg as a library if an example `rootLocation` and resulting path or something could be given. Like should it include the URI (s3://)? And will `snapshot_Id` still be appended after that? Some of these questions might be implementation specific, but it would be nice if the javadoc could clear up at least the expectated format of `rootLocation` without having to consult unit tests etc.

But I also recognize that there might be more javadocs to come (like for the corresponding Spark action) and some of those questions might be better answered elsewhere.

Not blocking but something to think about  🙂